src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+
+    for( auto itr( faces.begin()); itr !=faces.end(); ++itr)
+    { 
+      if     ( faceToEdges[*itr][0] == NO_ID )  faceToEdges[*itr][0] = edgeIndex ;
+      else if( faceToEdges[*itr][1] == NO_ID )  faceToEdges[*itr][1] = edgeIndex ;
+      else if( faceToEdges[*itr][2] == NO_ID )  faceToEdges[*itr][2] = edgeIndex ;
+      else if( faceToEdges[*itr][3] == NO_ID )  faceToEdges[*itr][3] = edgeIndex ;
+      else assert( false ); 
+    }
+  }
+  return true;
+}
+
+bool HexMeshConnectivityBuilder::computeEdgesToFaces(ArrayOfSets<localIndex> & edgeToFaces ) const
+{
+  // 1 - Allocate - Without counting  -- Should we overallocate a bit ?
+  // TODO Count for tet meshes one should count
+  int numberOfFacesAroundEdge = 5; // In a hexahedral mesh this is enough
+  edgeToFaces.resize( 0 );
+  edgeToFaces.reserve( nbEdges );
+  edgeToFaces.reserveValues( nbEdges * numberOfFacesAroundEdge );
+
+  // Append and set the capacity of the individual arrays.
+  for (int i = 0; i < nbEdges; ++i)
+  {
+    edgeToFaces.appendSet(numberOfFacesAroundEdge);
+  }
+
+  // 2 - Get the mapping
+  std::vector< localIndex > allFacesToUniqueFace;
+  computeAllFacesToUniqueFace( allFacesToUniqueFace );
+
+  for (unsigned int edgeIndex = 0; edgeIndex < m_uniqueEdges.size(); ++edgeIndex)
+  {
+    std::set<localIndex> faces;
+    getFacesAroundEdge( edgeIndex, allFacesToUniqueFace, faces );
+    
+    for( auto itr( faces.begin()); itr !=faces.end(); ++itr)
+    {
+      edgeToFaces.insertIntoSet(edgeIndex, *itr);
+    }
+  }
+  return true;
+}
+
+// Allocation is managed by the CellBlock 
+bool HexMeshConnectivityBuilder::computeElementsToFacesOfCellBlocks() 
+{
+  for(unsigned int curFace = 0; curFace < m_uniqueFaces.size(); ++curFace)
+  {
+    LocalFaceIndex id = m_uniqueFaces[curFace];
+
+    LocalCellIndex c = id / Hex::nbFacets;
+    HexFacetIndex face = id % Hex::nbFacets;
+    auto blockCell = getBlockCellFromManagerCell(c);
+
+    getCellBlock( blockCell.first ).setElementToFaces( blockCell.second, face, curFace );
+
+    LocalFaceIndex idNeighbor = m_allFacesToNeighbors[id];
+    if( idNeighbor != -1)
+    {
+      LocalCellIndex cNeighbor = idNeighbor / Hex::nbFacets;
+      HexFacetIndex faceNeighbor = idNeighbor % Hex::nbFacets;
+      auto blockCellNeighbor = getBlockCellFromManagerCell(cNeighbor) ;
+      getCellBlock( blockCellNeighbor.first ).setElementToFaces( blockCellNeighbor.second, faceNeighbor, curFace );
+    }
+  }
+
+  return true;
+}
+
+
+// Allocation is managed by the CellBlock 
+bool HexMeshConnectivityBuilder::computeElementsToEdgesOfCellBlocks() 
+{
+  for (unsigned int edgeIndex = 0; edgeIndex < m_uniqueEdges.size(); ++edgeIndex)
+  {
+    unsigned int first = m_uniqueEdges[edgeIndex];
+    unsigned int last = m_allEdges.size();
+
+    if( edgeIndex+1 < m_uniqueEdges.size() ) {
+      last = m_uniqueEdges[edgeIndex+1];
+    }
+
+    for(unsigned int i = first ; i < last; ++i)
```

Comment:
I'm a bug concerned by the naming since `last` is never reached (`last` does not mean `after the last`).
Maybe use the std wording `begin` and `end`?

Reply:
OK, I was stuck to my `begin`/`end` but it matches indeed so that's cool.